prboom2/src/SDL/i_system.c

```diff
@@ -285,24 +285,47 @@ const char* I_GetTempDir(void)
 }
 
 #else
-// cph - V.Aguilar (5/30/99) suggested return ~/.lxdoom/, creating
-//  if non-existant
-// cph 2006/07/23 - give prboom+ its own dir
-static const char prboom_dir[] = {"/.dsda-doom"}; // Mead rem extra slash 8/21/03
+
+static char *I_ConcatDir(const char *basedir, const char *subdir)
+{
+  char *buffer;
+  size_t len, sublen;
+
+  // Remove trailing slashes to be safe.
+  len = strlen(basedir);
+  while (len > 0 && basedir[len - 1] == '/')
```

Comment:
Do we need to be this paranoid versus the existing `if (base[len-1] == '/') base[len-1] = 0;` statement, if we are not really doing proper/full path validation to begin with?